### ISSUE TITLE
Correction issue due to backend LX buffer overflow

### ISSUE THREAD

**Issue by chichun-charlie-liu** [torch-spyre/torch-spyre#3222]
### What happened?

Compiled ops whose per-core LX scratchpad usage reaches **1600KB** (97.7% of the 1638.4KB / 80%-of-2MB frontend partition) produce silently corrupted output on device — no error or warning anywhere in the compile pipeline.

Through env var `DXP_LX_FRAC_AVAIL`, 0.2 by default, the frontend (torch-spyre) and backend (`dxp_standalone`) agree on a fixed partition of the 2MB per-core LX scratchpad: `[0, 1638.4KB)` for the frontend, `[1638.4KB, 2048KB)` for the backend. We observed that (with `DXP_VERBOSE=1`) the backend's own finalized schedule tree shows it placing an internal LX allocation at address **`1625344` bytes (1587.25KB)** - **overflown from backend's partition**. When a frontend buffer is still live at that address (which happens once per-core LX usage is high enough, empirically ~1600KB), the backend's allocation overwrites it mid-op.

We have not confirmed whether this address is fixed regardless of op/shape or happens to coincide across the cases tested.

### What did you expect to happen?

If additional buffer is needed beyond the partition boundary, a compilation error or alarm should be raised.

### How can we reproduce it?

```python
import torch, torch_spyre

def fn(x):
    return torch.mean(x * x, dim=-1) + 1e-8

torch.manual_seed(0xAFFE)
x_cpu = torch.randn(32, 512, HIDDEN, dtype=torch.float16)  # HIDDEN = 1536 or 1600
cpu_result = fn(x_cpu)
result = torch.compile(fn)(x_cpu.to(device="spyre")).to("cpu")
torch.testing.assert_close(result, cpu_result, atol=1e-3, rtol=1e-2)
```
**a) With `+ 1e-8`:** both `HIDDEN=1536` and `HIDDEN=1600` **fail** (10.4% and 6.2% mismatch respectively). Here `mean`'s output is consumed by `+eps` in the same fused kernel, so it also gets LX-pinned — pinned buffers: `mul` output (1536KB or 1600KB) + `mean` output (64KB), concurrently live, peak 1600KB or 1664KB either way ≥ the failing threshold.

**b) Without `+ 1e-8`** (`return torch.mean(x * x, dim=-1)`): `HIDDEN=1536` **passes**, `HIDDEN=1600` **fails**. With no downstream consumer, `mean`'s output is single-use and stays in HBM — only `mul`'s output is LX-pinned. So peak LX usage equals `mul`'s output size alone: 1536KB (passes) or 1600KB (fails). This isolates the bug to `mul`'s buffer alone reaching 1600KB, with no `mean`-LX involvement at all.

Also reproduces with a pure pointwise chain (no reduction): `y = -x; z =
y + y + 1e-8` at `hidden=1600`.

**How LX usage and the backend address were confirmed**

- Per-core LX pinning/peak usage: run with `SPYRE_INDUCTOR_LOG=1 SPYRE_INDUCTOR_LOG_LEVEL=DEBUG`, grep stderr for
  `scratchpad` — the `spyre.inductor.scratchpad.plan_solver` and  `.allocator` lines report exactly which buffers are LX-pinned and their  per-timestep resident size.
- Backend allocation addresses: run with `DXP_VERBOSE=1`. This produces `debug/sdsc_N/sdsc_N.out.out.out.json` under each op's directory in `<TORCHINDUCTOR_CACHE_DIR>/inductor-spyre/sdsc_*/debug/` — the `scheduleTree_` list in this file has an `allocate` node per buffer with a concrete `startAddressCoreCorelet_` address. The colliding address (`1625344`) was found this way in both the `mean` case and an unrelated pointwise `neg` case.

A full `/tmp/torchinductor_cliu/` capture from one failing run (`HIDDEN=1600`, with `+ 1e-8`, compiled with `DXP_VERBOSE=1`) is attached.  The relevant file is `inductor-spyre/sdsc_fused_add_mean_mul_0_*/debug/sdsc_1/sdsc_1.out.out.out.json` (the `mean` op), where `allocate_lds1_lx` sits at address `1625344`, inside `mul`'s still-live output buffer spanning `[0, 1600KB)`.

### Any environmental details we need to know?

pod/env was rebuilt on Jul13.

### Anything else we need to know?

A temporary guard is added in PR #3154, constraining the frontend max usage to 80%*2MB - 100KB. 

### Relevant log output

```Shell

```

**Comment by chichun-charlie-liu** [comment]
[sdsc_only.zip](https://github.com/user-attachments/files/30142736/sdsc_only.zip)
